### ISSUE TITLE
MacOS Catalina breaks arrow keys in file dialogs

### ISSUE THREAD

**Issue by jbclements** [racket/racket#2970]
It appears to me that under Catalina, arrow keys in the Mac file dialog don't work at all.

Steps to reproduce:

1) Start DrRacket (release candidate or 7.4, both behave the same).
2) Hit CMD-O (open file)
3) click once on a file in a directory with more than one file. (don't open, just select)
4) hit the up and down arrow keys, hear beep rather than seeing highlighted file change.

**Comment by jbclements** [comment]
I transferred this issue to the racket repo because I can confirm that it occurs without DrRacket.

1) put this in a file bad-file-dialog.rkt:
```` racket
#lang racket

(require racket/gui/base)

(get-file)

(get-file)
````

2) Run this file at the command-line in MacOS 10.15 Catalina. 
3) In the resulting file dialog, navigate to a directory with some files in it, click once to
select one of them, and notice that the up and down arrow keys don't change the selection.
4) Type cmd-g to open the "Go To the Folder" modal overlay.
5) Type cmd-. to cancel.
6) Type cmd-g again... see that the modal overlay doesn't open.
7) close the dialog.
8) See the next call to get-file open a dialog, notice that now cmd-g doesn't even work once.

**Comment by mflatt** [comment]
I haven't been able to make this work. The issue seems to be that the open and save dialogs are run as "out of process" or "sandbox mode" dialogs. Along those lines, it seems possible that declaring entitlements would make the dialog work, but that's a wild guess.

**Comment by jbclements** [comment]
To make sure that I understand you: when you say "haven't been able to make this work", you're indicating that you can reproduce the bug but not repair it, right?

Re: entitlements, I think that they may be enforced in January.

https://developer.apple.com/news/?id=09032019a

My optimistic (naive?) hope is that entitlements can be applied to binaries after the build process is complete; my sense is that Apple is pretty good about keeping the command-line workflow alive, although they certainly don't publicize it. Does this match your understanding?

If so, I should maybe look into what's required to make entitlements work.

**Comment by mflatt** [comment]
Yes, that's right: I see the behavior that you describe, but haven't found a repair.

**Comment by jbclements** [comment]
Possibly relevant; today on the console I saw this warning:

2020-01-04 12:15:32.159 DrRacket[21367:460479] WARNING: <NSSavePanel: 0x7f818ecb3830> found it necessary to prepare implicitly; please prepare panels using NSSavePanel rather than NSApplication or NSWindow.
2020-01-04 12:15:32.161 DrRacket[21367:460479] WARNING: <NSSavePanel: 0x7f818ecb3830> found it necessary to start implicitly; please start panels using NSSavePanel rather than NSApplication or NSWindow.
2020-01-04 12:15:32.161 DrRacket[21367:460479] WARNING: <NSSavePanel: 0x7f818ecb3830> running implicitly; please run panels using NSSavePanel rather than NSApplication.

Could this be related to this issue?

I quickly grepped the racket source, and found *no* occurrences of NSSavePanel or NSWindow, but the only instance of NSApplication I found was a part of NSApplicationMain.... so I'm probably missing something.

**Comment by melted** [comment]
The code is over in the gui repo, I think: https://github.com/racket/gui/blob/master/gui-lib/mred/private/wx/cocoa/filedialog.rkt

**Comment by jbclements** [comment]
Interesting. Well, maybe I should move this issue over there. Makes me feel like I'm accomplishing something?
